feat(graphql-rpc): Read optimistic transactions in graphql

tomxey opened the pull request:
# Description of change

Transaction effects are now available immediately after execution through `Query.transactionBlock`.
However, other queries that depend on the chain’s indexed state (e.g., address-level balance updates) may still lag until the transaction has been checkpointed.
To confirm that a transaction has been included in a checkpoint, user can query `Query.transactionBlock` and check whether the `effects.checkpoint` field is set (or `null` if not yet checkpointed).

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/8418

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [x] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by @iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [ ] Verification of API backward compatibility.

### Release Notes

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [x] GraphQL:
    Transaction effects are now available immediately after execution via `Query.transactionBlock`.
    Note that other queries relying on the chain’s indexed state (such as address-level balance updates) may still show a delay until the transaction is checkpointed.
    To verify checkpoint inclusion, query `Query.transactionBlock` and check the `effects.checkpoint` field — it will be `null` until the transaction has been checkpointed.
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

tomxey committed:
feat(graphql-rpc): Read optimistic transactions in graphql

tomxey committed:
Update snapshots

tomxey committed:
Regenerate TS files

kodemartin commented on crates/iota-graphql-rpc/src/types/transaction_block/mod.rs:
```diff
@@ -473,36 +477,75 @@ impl Loader<DigestKey> for Db {
+                        inner,
+                        checkpoint_viewed_at: u64::MAX,
+                    },
+                );
```
This duplicates logic. I would suggest `results.insert` after the `if ... else if` block.

tomxey replied:
Done in 77e8bf664c008e0f25a5e10d8ec7d46d919994fe

kodemartin reviewed (approved):
Looks good! Just a single minor suggestion.

tomxey committed:
Review remarks: have only one insert

tomxey committed:
Fallback to optimistic table in case checkpointed transaction is newer that checkpoint_viewed_at

marc2332 committed:
feat(ts-sdk): Add `waitMode` in `waitForTransaction` (#8828)

Based off https://github.com/iotaledger/iota/pull/8645

Allows to specify how to wait the transaction.


[run-ci]
